### ISSUE TITLE
Ability to show suggestion only when input value changes without needing to maintain state outside of auto-suggest itself

### ISSUE THREAD

**Issue by puspa** [moroshko/react-autosuggest#105]
The callback to method "showWhen" doesn't provide enough context like previous value. "showWhen" method thus can't make the call to show suggestions without maintaining its own state of last input value, which is less desirable. As "onInputFocus" also triggers the check, it can result in one extra n/w hop if the value is pre-populated as part of props.  

Is there certain approach we can take to handle this use case?
Can callback to "showWhen" method also forward previous value in addition to new value or does adding a new property to suggest only on input change makes sense?

Thanks for your help.

**Comment by moroshko** [comment]
@puspa Could you please describe your use case, so that I could understand why do you need the previous value?

**Comment by puspa** [comment]
Thank you for prompt reply. Following is my use case.
1. Autosuggest starts with pre-populated value. 
2. When user clicks on the autosuggest "onInputFocus" gets fired which eventually calls "showWhen".
3. " showWhen" at this point sees pre-populated value as valid user input and returns true. Which causes it to make a n/w request to get suggestions for the pre-populated value. This is not desirable as technically we already have value we wanted.

Basically I want it to make suggestions call(n/w or cached) only if user changes input. Trying to do that even with a wrapper implementatuon has proven difficult, hence the request for some means to achieve it.

Thanks!

**Comment by moroshko** [comment]
@puspa Looks like you are using react-autosuggest 2.x.

react-autosuggest 3.x has a better API, and I believe it will be easier to achieve what you want if you use 3.x.

Could you please create a Codepen that demonstrates your issue? You can start from [this Codepen](http://codepen.io/moroshko/pen/EPZpev) (it uses 3.3.2).

**Comment by moroshko** [comment]
No activity, closing for now.
